File: Api/Companies/iCompanyPortal.Api.Companies.Client/Subscription.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace iCompanyPortal.Api.Companies.Client
{
    public class Subscription
    {
        public int CompanyId { get; set; }

        public SubscriptionPlanType Type { get; set; }

        public SubscriptionLength Length { get; set; }

        public string CurrencyCode { get; set; }

        public DateTime SubscribedOn { get; set; }

        public DateTime ExpiresOn { get; set; }

        public SubscriptionStatus Status { get; set; }

        public Company Company { get; set; }

        public SubscriptionPlan SubscriptionPlan { get; set; }
    }
}

File: App/icompanyportal/projects/companies-api/src/lib/models/subscription.ts
import { SubscriptionLength } from './subscription-length';
import { SubscriptionPlanType } from './subscription-plan-type';
import { SubscriptionStatus } from './subscription-status';

export interface Subscription {
  companyId?: number;
  type?: SubscriptionPlanType;
  length?: SubscriptionLength;
  currencyCode?: string;
  subscribedOn?: string;
  expiresOn?: string;
  status?: SubscriptionStatus;
}
